From 0a6fc69d18c80b32e88c5fc00eaeba9bb8b01407 Mon Sep 17 00:00:00 2001
From: Takuro Ashie <ashie@clear-code.com>
Date: Tue, 11 Mar 2025 13:16:05 +0900
Subject: [PATCH 1/7] Define missing MAXNS & MAXDNSRCH

```
In file included from qrap.c:36:
./passt.h:130:21: error: use of undeclared identifier 'MAXNS'
  130 |         struct in_addr dns[MAXNS + 1];
      |                            ^
./passt.h:171:22: error: use of undeclared identifier 'MAXNS'
  171 |         struct in6_addr dns[MAXNS + 1];
      |                             ^
./passt.h:270:25: error: use of undeclared identifier 'MAXDNSRCH'
  270 |         struct fqdn dns_search[MAXDNSRCH];
      |                                ^
```
---
 passt.h | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/passt.h b/passt.h
index 0dd4efa..8fa4e74 100644
--- a/passt.h
+++ b/passt.h
@@ -82,6 +82,12 @@ extern char *epoll_type_str[];
 	                                    epoll_type_str[(n)] : "?")
 
 #include <resolv.h>	/* For MAXNS below */
+#ifndef MAXNS
+#  define MAXNS 3
+#endif /* MAXNS */
+#ifndef MAXDNSRCH
+#  define MAXDNSRCH 6
+#endif /* MAXDNSRCH */
 
 /**
  * struct fqdn - Representation of fully-qualified domain name
-- 
2.34.1

